src/common/types.cc

```diff
@@ -22,11 +22,15 @@
 #define UINT8_MAX (255)
 #endif
 
-const shard_id_t shard_id_t::NO_SHARD(UINT8_MAX);
+#ifndef INT8_MIN
+#define INT8_MIN (-128)
+#endif
+
+const shard_id_t shard_id_t::NO_SHARD(INT8_MIN);
```

Comment:
This change breaks upgrades.  Starting wip-recovery-temp built OSD crashes on older ObjectStore.  Also, breaks import-rados feature of ceph-objectstore-tool because shard_id inside spg_t inside of export of replicated pg, would no longer match the new NO_SHARD.